go/enclave/storage/init/sqlite/001_init.sql

```diff
@@ -61,62 +68,77 @@ create table if not exists batch
+    body           int       NOT NULL REFERENCES batch_body,
+    l1_proof_hash  binary(32),
+    l1_proof       INTEGER, -- normally this would be a FK, but there is a weird edge case where an L2 node might not have the block used to create this batch
+    is_executed    boolean   NOT NULL
     --   the unique constraint is commented for now because there might be multiple non-canonical batches for the same height
 --   unique (height, is_canonical, is_executed)
-    );
+);
 create index IDX_BATCH_HASH on batch (hash);
 create index IDX_BATCH_HEIGHT on batch (height, is_canonical);
 create index IDX_BATCH_Block on batch (l1_proof);
 
 create table if not exists tx
 (
-    hash           binary(16) primary key,
+    id             INTEGER PRIMARY KEY AUTOINCREMENT,
+    hash           binary(4),
     full_hash      binary(32) NOT NULL,
     content        mediumblob NOT NULL,
     sender_address binary(20) NOT NULL,
     nonce          int        NOT NULL,
     idx            int        NOT NULL,
     body           int REFERENCES batch_body
-    );
+);
+create index IDX_Tx_HASH on tx (hash);
 
 create table if not exists exec_tx
 (
-    id                       binary(16) PRIMARY KEY, -- batch_hash||tx_hash
-    created_contract_address binary(20),
-    receipt                  mediumblob,
+    id                            INTEGER PRIMARY KEY AUTOINCREMENT,
+    created_contract_address      binary(4),
+    created_contract_address_full binary(20),
+    receipt                       mediumblob,
     --     commenting out the fk until synthetic transactions are also stored
---     tx                       binary(16) REFERENCES tx,
-    tx                       binary(16) NOT NULL,
-    batch                    int        NOT NULL REFERENCES batch
-    );
-create index IX_EX_TX1 on exec_tx (tx);
+    tx                            INTEGER,
+    batch                         INTEGER NOT NULL REFERENCES batch
+);
+create index IDX_EX_TX_TX on exec_tx (tx);
+create index IDX_EX_TX_BATCH on exec_tx (batch);
+create index IDX_EX_TX_CCA on exec_tx (created_contract_address);
 
 -- todo denormalize. Extract contract and user table and point topic0 and rel_addreses to it
 create table if not exists events
 (
-    topic0          binary(32) NOT NULL,
-    topic1          binary(32),
-    topic2          binary(32),
-    topic3          binary(32),
-    topic4          binary(32),
-    datablob        mediumblob,
-    log_idx         int        NOT NULL,
-    address         binary(20) NOT NULL,
-    lifecycle_event boolean    NOT NULL,
-    rel_address1    binary(20),
-    rel_address2    binary(20),
-    rel_address3    binary(20),
-    rel_address4    binary(20),
-    exec_tx_id      binary(16) REFERENCES exec_tx
-    );
+    topic0            binary(4)  NOT NULL,
+    topic1            binary(4),
+    topic2            binary(4),
+    topic3            binary(4),
+    topic4            binary(4),
+    topic0_full       binary(32) NOT NULL,
```

Comment:
Minor thing (subjective) but might be a bit more obvious if we call the `_full` versions by the actual name and use a convention for the short hashes instead, like `topic0_lookup` or something.

Reply:
I'd love to do this, but it's a major PITA. The replaces would be super fiddly